qml/MainPage.qml

```diff
@@ -13,38 +13,104 @@ Page {
             return new RegExp('');
         }
     }
+    property bool searchMode: false
 
     header: PageHeader {
+        id: defaultHeader
         title: i18n.tr("Lomiri Touch Logs")
         flickable: scrollView.flickableItem
+    }
+
+    states: [
+        State {
+          id: defaultState
+          when: !mainPage.searchMode
 
-        trailingActionBar.actions: [
-        Action {
-            text: i18n.tr("Settings")
-            onTriggered: mainView.showSettings()
-            iconName: "settings"
+          property list<QtObject> trailingActions: [
+              Action {
+                  text: i18n.tr("Settings")
+                  onTriggered: mainView.showSettings()
+                  iconName: "settings"
+              },
+              Action {
+                  text: i18n.tr("About")
+                  onTriggered: pStack.push(Qt.resolvedUrl("AboutPage.qml"))
+                  iconName: "info"
+              },
+              Action {
+                  iconName: "search"
+                  text: i18n.tr("Search")
+                  onTriggered: {
+                      mainPage.searchMode = true
+                      //mainPage.header = searchHeader
+                      searchField.forceActiveFocus()
+                  }
+              }
+          ]
+
+          PropertyChanges {
+              target: defaultHeader
+              trailingActionBar.actions: defaultState.trailingActions
+              leadingActionBar.actions:  []
+          }
         },
-        Action {
-            text: i18n.tr("About")
-            onTriggered: pStack.push(Qt.resolvedUrl("AboutPage.qml"))
-            iconName: "info"
+
+        State {
+            id: searchState
+            when: mainPage.searchMode
+
+            property list<QtObject> leadingActions: [
```

Comment:
This is basically a constant that is only used once, I think it would be better to just define this array when assigning in `PropertyChanges` below

Reply:
not working apparently, QML does not allow to create QML Component in PropertyChange

Reply:
Huh, didn't expect that, I wonder if it's because this is a UITK thing